### PR TITLE
Fix copy constructor and assignment for ov::Any

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
 - Create custom copy constructor and copy assignment operator for ov::Any
 - Add test
 - Remove clone_map methods as excess
 - Fix cmake in gna legacy tests

### Tickets:
 - CVS-106375

### CODE REVIEW

**Comment by olpipi** [comment]
Hi @ilyachur @ilya-lavrenov 
Could you please take a look?
Actual implementation of ov::Any does not allow to copy value inside ov::Any. We need to change this behavior. 
I implemented a custom copy for ov::Any.

**Comment by ilya-lavrenov** [comment]
Can we also remove these WAs https://github.com/search?q=repo%3Aopenvinotoolkit%2Fopenvino%20clone_map&type=code ?

File: src/core/src/any.cpp [hunk 1]
```diff
@@ -73,6 +73,18 @@ Any::~Any() {
...
 
+Any::Any(const Any& other) {
+    *this = other;
+};
+
+Any& Any::operator=(const Any& other) {
+    if (other._temp)
+        _temp = other._temp->copy();
+    if (other._impl)
+        _impl = other._impl->copy();
```
Comment (ilya-lavrenov) [inline src/core/src/any.cpp:84]: what's about `_so` ?

Comment (olpipi) [reply src/core/src/any.cpp:84]: fixed

File: src/core/include/openvino/core/model.hpp [hunk 2]
```diff
@@ -240,7 +248,12 @@ class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
     /// when evaluating the model. This additional information can be shared across nodes.
     bool evaluate(ov::TensorVector& output_tensors,
                   const ov::TensorVector& input_tensors,
-                  ov::EvaluationContext evaluation_context = ov::EvaluationContext()) const;
+                  ov::EvaluationContext& evaluation_context) const;
+
+    /// \brief Evaluate the model on inputs, putting results in outputs.
+    /// \param output_tensors Tensors for the outputs to compute. One for each result
+    /// \param input_tensors Tensors for the inputs. One for each inputs.
+    bool evaluate(ov::TensorVector& output_tensors, const ov::TensorVector& input_tensors) const;
```
Comment (ilya-lavrenov) [inline src/core/include/openvino/core/model.hpp:256]: how these changes are related to ov::Any?

Comment (olpipi) [reply src/core/include/openvino/core/model.hpp:256]: Hi
Previously Model::evaluate took evaluation_context by value. It works only if ov:Any has broken copy constructor.
Model::evaluate is broken due to ov::Any copy fix.

So I changed Model::evaluate declaration to take evaluation_context  by reference.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Review by mryzhov — APPROVED** [review]
Looks fine from GNA perspective

**Pull request merged**